ca_on_ajax/people.py

```diff
@@ -1,49 +1,43 @@
-        assert len(councillors), 'No councillors found'
+        councillors = page.xpath('//table[@class="councilTable"]')
         for councillor in councillors:
-            name = councillor.text_content()
+            image = councillor.xpath('.//@src')[0]
+            councillor_name = councillor.xpath('.//tr/td[1]/p[1]/img/@alt')
+            assert len(councillors), 'No councillors found'
 
-            url = councillor.attrib['href']
-            page = self.lxmlize(url)
+            for i in councillor_name:
+                name = i.split('-', 1)[0].strip()
+                district = i.split('Councillor ')[-1].strip()
 
-            if councillor == councillors[0]:
+            if name == councillor_name[0]:
                 district = 'Ajax'
                 role = 'Mayor'
+
             else:
-                district = re.findall(r'Ward.*', page.xpath('//div[@id="printAreaContent"]//h1')[0].text_content())[0].strip()
-                role = page.xpath('//div[@id="printAreaContent"]//h1')[0].text_content()
-                role = re.search('((?:Regional )?Councillor)', role).group(1)
+                role = 'Councillor'
 
-            name = name.replace(role, '').strip()
+            cell = councillor.xpath('.//p[contains(.,"Cel")]/text()')[0]
+            tel = councillor.xpath('.//p[contains(.,"Cel")]/text()')[1]
+            phone = cell.replace('\xa0', ' ')
 
-            p = Person(primary_org='legislature', name=name, district=district, role=role)
+            p = Person(primary_org='legislature',
+                       name=name,
+                       district=district,
+                       role=role)
             p.add_source(COUNCIL_PAGE)
-            p.add_source(url)
-
-            p.image = page.xpath('//div[@class="intQuicklinksPhoto"]//img/@src')[0]
-
-            contact_info = page.xpath('//table[@class="datatable"][1]//tr')[1:]
-            for line in contact_info:
-                contact_type = line.xpath('./td')[0].text_content().strip()
-                if re.match(r'(Home)|(Cell)|(Phone)|(Fax)|(Email)', contact_type):
-                    contact = line.xpath('./td')[1].text_content().strip()
-                    contact_type = CONTACT_DETAIL_TYPE_MAP[contact_type]
-                    p.add_contact(contact_type, contact, '' if contact_type == 'email' else 'legislature')
-                elif contact_type == 'Address':
-                    contact = ''.join(line.xpath('./td[2]//text()')).strip()
-                    p.add_contact(contact_type, contact, 'legislature')
-                else:
-                    contact = line.xpath('./td[2]/a/@href')[0]
-                    p.add_link(contact)
+            p.image = image
+
+            if phone:
+                p.add_contact('cell', phone, 'legislature')
+            if tel:
+
+                p.add_contact('email', self.get_email(councillor))
```

Comment:
I don't understand the logic… if you file a telephone number, you add an email? What happens to the telephone number? Also, please remove the newline after the if-condition.